hackathon_site/hardware/test_api.py

```diff
@@ -0,0 +1,97 @@
+from rest_framework.test import APITestCase
+from django.urls import reverse
+from rest_framework import status
+from event.models import User
+from hardware.models import Hardware, Category
+from hardware.serializers import HardwareSerializer, CategorySerializer
+
+
+class HardwareListViewTestCase(APITestCase):
+    def setUp(self):
+        self.password = "foobar123"
+        self.user = User.objects.create_user(
+            username="foo@bar.com",
+            password="foobar123",
+            first_name="Test",
+            last_name="Bar",
+            email="foo@bar.com",
+        )
+        self.hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            quantity_available=4,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+        self.hardware_serializer = HardwareSerializer(self.hardware)
+        self.view = reverse("api:hardware:hardware-list")
+
+    def _login(self):
+        self.client.login(username=self.user.username, password=self.password)
+
+    def test_user_not_logged_in(self):
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_category_success(self):
+        self._login()
+        expected_response = {
+            "id": 1,
+            "name": "name",
+            "categories": [],
+            "model_number": "model",
+            "manufacturer": "manufacturer",
+            "datasheet": "/datasheet/location/",
+            "quantity_available": 4,
+            "quantity_remaining": 4,
+            "notes": None,
+            "max_per_team": 1,
+            "picture": "http://testserver/api/hardware/hardware/picture/location",
+        }
+
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.json()
+
+        self.assertEqual(expected_response, data["results"][0])
+
+
+class CategoryListViewTestCase(APITestCase):
+    def setUp(self):
+        self.password = "foobar123"
```

Comment:
Since we're going to have to setup the user and login method for every test ever, it would make sense to define a mixin for that somewhere, perhaps in `hackathon_site/tests.py`. Something like:

```python
class SetupUserMixin:
    def setUp(self):
        self.password = "foobar123"
        self.user = User.objects.create_user(
            username="foo@bar.com",
            password="foobar123",
            first_name="Test",
            last_name="Bar",
            email="foo@bar.com",
        )

    def _login(self):
        self.client.login(username=self.user.username, password=self.password)
```

Then this test class would become eg:

```python
class CategoryListViewTestCase(SetupUserMixin, APITestCase):
    def setUp(self):
        super().setUp()
        
        self.category = ...
```